packages/core/src/asset/mintDrop.ts

```diff
@@ -0,0 +1,108 @@
+import { Interface } from '@ethersproject/abi'
+import type { BigNumberish, Signer } from 'ethers'
+import { BigNumber } from 'ethers'
+import invariant from 'ts-invariant'
+import { pollOwnership } from '../exchange/offerQuantityChanges'
+import type { Sdk } from '../graphql'
+import type { Address, ChainId, IState, TransactionHash, UUID } from '../types'
+import { toAddress, toTransactionHash } from '../utils/convert'
+import { sendTransaction } from '../utils/transaction'
+
+export type State =
+  | IState<'TRANSACTION_SIGNATURE', {}>
+  | IState<'TRANSACTION_PENDING', { txHash: TransactionHash }>
+  | IState<'OWNERSHIP', {}>
+
+const abi = [
+  {
+    anonymous: false,
+    inputs: [
+      {
+        indexed: true,
+        internalType: 'address',
+        name: 'from',
+        type: 'address',
+      },
+      {
+        indexed: true,
+        internalType: 'address',
+        name: 'to',
+        type: 'address',
+      },
+      {
+        indexed: true,
+        internalType: 'uint256',
+        name: 'tokenId',
+        type: 'uint256',
+      },
+    ],
+    name: 'Transfer',
+    type: 'event',
+  },
+]
+const transferInterface = new Interface(abi)
+
+export async function mintDrop(
+  sdk: Sdk,
+  dropId: UUID,
+  quantity: BigNumberish,
+  signer: Signer,
+  onProgress?: (state: State) => void,
+): Promise<
+  {
+    chain: ChainId
+    collection: Address
+    token: string
+  }[]
+> {
+  const address = await signer.getAddress()
+
+  const { drop } = await sdk.FetchDrop({ dropId })
+  invariant(drop, "Can't find drop")
+
+  const {
+    createDropMintTransaction: { transaction },
+  } = await sdk.CreateDropMintTransaction({
+    dropId,
+    quantity: BigNumber.from(quantity).toString(),
+    minter: toAddress(address),
+  })
+
+  onProgress?.({ type: 'TRANSACTION_SIGNATURE', payload: {} })
+  const tx = await sendTransaction(signer, transaction)
+
+  onProgress?.({
+    type: 'TRANSACTION_PENDING',
+    payload: { txHash: toTransactionHash(tx.hash) },
+  })
+  const receipt = await tx.wait()
+  const transfers = [] as { tokenId: string }[]
+  for (const log of receipt.logs || []) {
+    try {
+      const parsed = transferInterface.parseLog(log)
+      if (parsed.name !== 'Transfer') continue
+      transfers.push({ tokenId: parsed.args.tokenId.toString() })
+    } catch (e) {}
+  }
+
+  invariant(transfers[0], 'Error checking ownership')
+
+  onProgress?.({ type: 'OWNERSHIP', payload: {} })
+  await pollOwnership(
+    sdk,
+    drop.chainId,
+    drop.collectionAddress,
+    transfers[0].tokenId,
+    toAddress(address),
+    BigNumber.from(0),
```

Comment:
I believe this will only work when minting the first time, if we mint again, we might have an issue. I wanted to test this flow but I can't according to this comment https://github.com/liteflow-labs/starter-kit/pull/359#issuecomment-1702881299

Reply:
Let's go with that, from the tests it seems that this is working well, so let's finalize this and if needed do another version if there is indeed an issue

Reply:
Note: this should actually "work" but would be a false positive with the following flow:

- User mints 1 token
  - The current ownership is 0
  - The asset is synced with the API
  - The new ownership is 1
  - The polling has a change between 0 and 1, so all is good
- User mints 1 more token
  - The current ownership is 1
  - The polling from 0 to 1 stops because it is different <-- This is the false positive
  - The asset is synced with the API
  - The new ownership is 2

Because of that, we would exit the function too early and might have issues with redirecting to the asset page, and the asset is not yet synced with the API, which will result in a 404

Reply:
moved into a dedicated issue https://github.com/liteflow-labs/liteflow-js/issues/179